### PR TITLE
Updating the package to include the work done by goul on the smarthub2

### PR DESCRIPTION
**Pull request by jxwolstenholme**
Reorganised most of the code to provide a cleaner method of differentiating between the smarthub 1 and 2

### CODE REVIEW

File: btsmarthub_devicelist/__init__.py [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+    for label in labels:
+        source = source.replace(label + ":", "\"" + label + "\":")
+
+    return source
+
+
+DEFAULT_IP = '192.168.1.254'
+
+
+class BTSmartHub(object):
```
Comment (d0ugal) [inline btsmarthub_devicelist/__init__.py:32]: Moving everything into a class means that the Home Assistant code will need to be updated. https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/bt_smarthub/device_tracker.py#L83-L84

File: btsmarthub_devicelist/__init__.py [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+    """ Represents a session to a BT SmartHub Router."""
+
+    def __init__(self, router_ip=None, smarthub_model=None):
```
Comment (d0ugal) [inline btsmarthub_devicelist/__init__.py:35]: You could use `router_ip=DEFAULT_IP` and remove the `if router ip:....` below

File: btsmarthub_devicelist/__init__.py [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+            self.smarthub_model = smarthub_model
+
+    def get_devicelist(self, only_active_devices=None):
+
+        if only_active_devices is None:
+            only_active_devices = True
+        elif only_active_devices:
+            only_active_devices = True
+        else:
+            only_active_devices = False
```
Comment (d0ugal) [inline btsmarthub_devicelist/__init__.py:55]: To ensure it is a boolean, you could replace this if/elif/else with `only_active_devices = bool(only_active_devices)`

**Comment by d0ugal** [comment]
Just had a quick look. I'd be happy to do a full code review if that would be useful!

File: btsmarthub_devicelist/__init__.py (hunk 1 above)
Comment (jxwolstenholme) [reply btsmarthub_devicelist/__init__.py:32]: Will you give this a go?
[https://github.com/jxwolstenholme/home-assistant/blob/dev/homeassistant/components/bt_smarthub/device_tracker.py](https://github.com/jxwolstenholme/home-assistant/blob/dev/homeassistant/components/bt_smarthub/device_tracker.py)

**Comment by jxwolstenholme** [comment]
Thanks @d0ugal - comments always appreciated

File: btsmarthub_devicelist/__init__.py (hunk 1 above)
Comment (d0ugal) [reply btsmarthub_devicelist/__init__.py:32]: I can try and test this over the next few days.

**Pull request merged**